content/posts/2022-09-14-introducing-jaiminho/index.md

```diff
@@ -0,0 +1,136 @@
+Be careful with this approach, **if any execution fails, the relayer will get stuck**. Otherwise, it would not possible to guarantee delivery order.  
+
+### Publish on commit
+
+This strategy will always execute the decorated function after the current transaction commit. With this approach, we don’t depend on an event relayer to execute the decorated function and deliver the message. Failed items will only be executed through the relayer. Although we can decrease the delay to execute the decorated function with this approach, **we cannot guarantee delivery order**.
+
+## Additional settings
+
+Besides the publish strategies, there are two other options that can be configured:
+
+### PERSIST_ALL_EVENTS
+
+While the default behavior of Jaiminho makes losing an event less likely, it still has a point of failure: if the notifying method fails and Jaiminho receives an error trying to store the database entry, the event will be lost. In order to avoid this possibility, the `PERSIST_ALL_EVENTS` configuration can be set to True, which will save the database entry *before* trying to notify the event. In this case, the notifying method will be called when the database transaction is committed, which will ensure that the event is persisted. This is not applicable when the publish strategy is **Keep order**, because all the messages must be stored to achieve this strategy.
+
+### DELETE_AFTER_SEND
+
+The `PERSIST_ALL_EVENTS` configuration can be combined with the `DELETE_AFTER_SEND` one. If `DELETE_AFTER_SEND` is activated, after a successful notifying of the event, the persisted database entry will be deleted, in order to make sure Jaiminho will not store successfully sent events.
+
+You may be asking: why isn’t the `PERSIST_ALL_EVENTS` configuration the default behavior of Jaiminho? The reason behind it is performance. For each event sent, the activation of this setting will cause two extra database operations (one insertion and one deletion), which can be a lot for use cases where there is a large number of events being sent. So, it’s up to your use case and you to decide whether you prefer the performance of just going to the database if something fails or the consistency guarantee of not losing any events.
+
+## Django Commands
+
+Jaiminho comes with two Django Commands that can be useful:
+
+### Event Relaying
+
+The Event Relaying Command will query the database for all events that were not sent in a given timebox and will try to re-send them using the original method, message and kwargs. Note that if the event relay runs after a deploy that changed or removed the original method, it will not be able to re-send the failed events and they will have to be dealt with manually. Also, the `DELETE_AFTER_SEND` configuration also applies to this command.
+
+Tip: Schedule this command to be regularly executed and Jaiminho will ensure all failed events are re-sent.
```

Comment:
Would be nice to include the consumer mode when the relayer runs forever (no scheduling)

Reply:
Done! Can you check the new text to see if it's well explained?